### DISCUSSION TITLE
Ignoring folders in home dir (MacOS)

### DISCUSSION THREAD

**Discussion by fiechr** [junegunn/fzf#3922] [Q&A]
I'm using `fd` with `fzf` and want it to ignore the `~/Library/` and `~/Applications/` folders, when I start a search inside the root of my home directory.

For this, I've created a .fdignore file in my home directory:
```
$ cat ~/.fdignore
Applications/
Library/
```
There's also a .config/fd/ignore:
```
$ cat .config/fd/ignore
.git/
```

Now, if I type "fzf" in my home dir, it works, and both directories are omitted.

But if I use the Ctrl + T shortcut or the command + "**" + Tab syntax the whole home directory is indexed.

Here's some more sys info:
```
$ zsh --version
zsh 5.9 (arm-apple-darwin23.0.0)
$ omz version
master (5b37e21)
$ brew --version
Homebrew 4.3.9
$ fzf --version
0.54.0 (brew)
$ which fzf
/opt/homebrew/bin/fzf
$ fd --version
fd 10.1.0
$ which fd
/opt/homebrew/bin/fd
$ echo $FZF_DEFAULT_COMMAND
fd --type f --hidden --exclude .git
```

Maybe it's related to the fzf plugin comming with Oh My ZSH?
https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/fzf

Any ideas?

**Comment by junegunn** [comment]
Please read https://github.com/junegunn/fzf?tab=readme-ov-file#environment-variables

**Reply by fiechr** [reply]
Thank you for your reply and link.

I think I figured it out now.

I've added this to my .zshrc and it to works now:

```
# fzf, use fd instead of find (fd reads ~/.fdignore)

export FZF_DEFAULT_COMMAND='fd --type f --hidden --exclude .git'

# Ctrl + T command
export FZF_CTRL_T_COMMAND=$FZF_DEFAULT_COMMAND

# Alt + C command (ESC + C on MacOS)
export FZF_ALT_C_COMMAND='fd --type d --hidden --exclude .git'

# "**" command syntax
_fzf_compgen_path() {
  fd --hidden --follow --exclude ".git" . "$1"
}

# "**" command syntax (for directories only)
_fzf_compgen_dir() {
  fd --type d --hidden --follow --exclude ".git" . "$1"
}
```
